src/main/java/org/opensearch/security/configuration/SecurityConfigDiffCalculator.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.configuration;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.security.DefaultObjectMapper;
+import org.opensearch.security.configuration.SecurityConfigVersionDocument.HistoricSecurityConfig;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+
+import com.flipkart.zjsonpatch.JsonDiff;
+
+/**
+ * Utility class to compute differences between two versions of security configurations
+ *
+ * @opensearch.experimental
+ */
+
+public class SecurityConfigDiffCalculator {
+    private static final Logger LOGGER = LogManager.getLogger(SecurityConfigDiffCalculator.class);
+
+    private static final ObjectMapper objectMapper = DefaultObjectMapper.objectMapper;
+
+    public static boolean hasSecurityConfigChanged(
+        Map<String, HistoricSecurityConfig<?>> oldConfig,
+        Map<String, HistoricSecurityConfig<?>> newConfig
+    ) {
+        try {
+            if (oldConfig == null || oldConfig.isEmpty()) {
+                LOGGER.info("Old configuration is empty. Treating as a new configuration.");
+                return true;
+            }
+
+            JsonNode oldNode = buildConfigDataNode(oldConfig);
+            JsonNode newNode = buildConfigDataNode(newConfig);
+
+            JsonNode diff = JsonDiff.asJson(oldNode, newNode);
+
+            if (diff.isEmpty()) {
+                LOGGER.info("No changes detected in security configuration.");
+                return false;
+            } else {
+                LOGGER.info("Detected changes in security configuration: {}", diff.toString());
+                return true;
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error while comparing security configurations", e);
```

Comment:
Any reason for ignoring errors ? 
This means it would not report as a 500 to the end user.